docs/preview/02-Features/02-message-handling/03-event-hubs.md

```diff
@@ -0,0 +1,131 @@
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.EventHubs;
+using Microsoft.Extensions.Logging;
+
+public class OrdersMessageHandler : IAzureEventHubsMessageHandler<Order>
+{
+    private readonly ILogger _logger;
+
+    public OrdersMessageHandler(ILogger<OrdersMessageHandler> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task ProcessMessageAsync(
+        Order orderMessage, 
+        AzureEventHubsMessageContext messageContext, 
+        MessageCorrelationInfo correlationInfo, 
+        CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Processing order {OrderId} for {OrderAmount} units of {OrderArticle} bought by {CustomerFirstName} {CustomerLastName}", orderMessage.Id, orderMessage.Amount, orderMessage.ArticleNumber, orderMessage.Customer.FirstName, orderMessage.Customer.LastName);
+
+        // Process the message.
+
+        _logger.LogInformation("Order {OrderId} processed", orderMessage.Id);
+    }
+}
+```
+
+## Message handler registration
+Once the message handler is created, you can very easily register it:
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+public class Program
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Arcus secret store will be used to lookup the connection strings, 
+        // for more information about the Arcus secret store see: https://security.arcus-azure.net/features/secret-store
+        services.AddSecretStore(stores => ...);
+
+        // Add Azure EventHubs message pump and use OrdersMessageHandler to process the messages.
+        services.AddEventHubsMessagePump("<my-eventhubs-name>", "Arcus_EventHubs_ConnectionString", "<my-eventhubs-blob-storage-container-name>", "Arcus_EventHubs_Blob_ConnectionString")
+                .WithEventHubsMessageHandler<OrdersMessageHandler, Order>();
+
+        // Note, that only a single call to the `.WithEventHubsMessageHandler` has to be made when the handler should be used across message pumps.
+    }
+}
+```
+
+The Azure EventHubs uses the `EventProcessorClient` internally. To learn more about this way of consuming messages from Azure EventHubs, see [Microsoft's official documentation](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/messaging.eventhubs.processor-readme).
+
+In this example, we are using the Azure EventHubs message pump to process event messages and use the connection strings stored in the Arcus secret store:
+- Azure EventHubs name: The name of the Event Hub that the processor is connected to, specific to the EventHubs namespace that contains it.
+- Azure EventHubs connection string secret name: The name of the secret to retrieve the Azure EventHubs connection string using your registered Arcus secret store implementation.
+- Azure EventHubs Blob storage container name: The name of the Azure Blob storage container in the storage account to reference where the event checkpoints will be stored and the load balanced.
```

Comment:
```suggestion
- Azure EventHubs Blob storage container name: The name of the Azure Blob storage container in the storage account to reference where the event checkpoints will be stored. The events will be streamed to this storage so that the client only has to worry about event processing, not event capturing.
```